cider/exec/operator/aggregate/CiderAggTargetColExtractorTmpl.h

```diff
@@ -129,7 +129,32 @@ class SimpleAggExtractor<ST, VarCharPlaceHolder> : public CiderAggTargetColExtra
   }
 
   void extract(const std::vector<const int8_t*>& rowAddrs, CiderBatch* output) override {
-    UNREACHABLE();
+    size_t rowNum = rowAddrs.size();
+    auto varcharOutput = output->asMutable<VarcharBatch>();
+
+    CHECK(varcharOutput->resizeBatch(rowNum));
+    uint8_t* buffer = varcharOutput->getMutableRawData();
+    int32_t* offset = varcharOutput->getMutableRawOffset();
+    uint8_t* nulls = varcharOutput->getMutableNulls();
+
+    offset[0] = 0;
+    int64_t null_count = 0;
+    for (size_t i = 0; i < rowNum; ++i) {
+      const int8_t* rowPtr = rowAddrs[i];
+      const int64_t* id = reinterpret_cast<const int64_t*>(rowPtr + offset_);
+      CiderByteArray value = hasher_->lookupValueById(*id);
+      // TODO(yizhong): null should not be decided by length
+      if (!value.len) {
```

Comment:
What will happen with insert an entry with `null` key? May be the hasher id is `-1`?

Reply:
Well, there is no case to return hasher id `-1` yet, and `null` will be also regarded as a `UStringVal`. Null case will be handled after we supported `null` in `CiderByteArray`.

Reply:
I think you can align @jikunshang about this special semantics.  He defined some in his [PR](https://github.com/intel/BDTK/pull/129/files).

Reply:
see [here](https://github.com/intel/BDTK/pull/129/files#diff-09df0308139f7a448df0a49f37bcc36ed82236515d665834cef9cb14f627fc28R122)